evalml/model_understanding/graphs.py

```diff
@@ -555,3 +555,31 @@ def graph_prediction_vs_actual(y_true, y_pred, outlier_threshold=None):
                                 marker=_go.scatter.Marker(color=color),
                                 name=name))
     return _go.Figure(layout=layout, data=data)
+
+
+def graph_prediction_vs_target_over_time(pipeline, X, y, dates):
+    """Plot the target values and predictions against time on the x-axis.
+
+    Arguments:
+        pipeline (PipelineBase): Fitted time series regression pipeline.
+        X (pd.DataFrame): Features used to generate new predictions.
+        y (pd.Series): Target values to compare predictions against.
+
+    Returns:
+        plotly.Figure showing the prediction vs actual over time.
```

Comment:
@freddyaboulton could we separate this into a data generation method and a graphing method? That way callers with a different UI can use the data generation method, and people who want plotly figs can use this method.

For this particular plot the data generated is fairly simple, just three vectors: predictions, actual and dates

Reply:
Yep I thought about doing that at first but thought it wasn't worth it since it was a one-liner. I'll push this up now!